elyra/util/git.py

```diff
@@ -0,0 +1,102 @@
+                 repo: str,
+                 branch: Optional[str] = None,
+                 server_url: Optional[str] = "https://api.github.com"):
+        """
+        Creates a Github Client for Elyra
+        :param token: Personal Access Token for use with Github
+                      See https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+        :param repo: Github Repository to use. Use Form : [Github Username/Org]/[Repository Name]
+                     e.g. elyra/examples
+        :param branch: Branch in the 'repo' to use. If not provided, this will use the default branch configured in the
+                       target repository
+        :param server_url:  Github API endpoint to use for the client. This is can be an Enterprise
+                          Github instance. By default the client will attempt to connect to the main Github API at
+                          https://api.github.com'
+        """
+
+        super().__init__()
+
+        self.server_url = server_url
+        self.repo_name = repo
+        self.branch = branch
+        self.client = Github(login_or_token=token, base_url=self.server_url)
+
+        try:
+            self.github_repository = self.client.get_repo(self.repo_name)
+        except GithubException as e:
+            self.log.error(f'Error accessing repository {self.repo_name}: ' + str(e))
+            raise RuntimeError(f'Error accessing repository {self.repo_name}: {str(e)}') from e
+
+    def upload_dag(self,
+                   pipeline_filepath: str,
+                   pipeline_name: str) -> None:
+        """
+        Push a DAG to a remote Github Repository
+        :param pipeline_filepath: filepath to the location of the DAG in the local filesystem
+        :param pipeline_name: the name of the file to be created in the remote Github Repository
+        :return:
+        """
+
+        try:
+            # Upload to github
+            with open(pipeline_filepath) as input_file:
+                content = input_file.read()
+
+                self.github_repository.create_file(path=pipeline_name + ".py",
+                                                   message="Pushed DAG " + pipeline_name,
+                                                   content=content,
+                                                   branch=self.branch)
+
+            self.log.info('Pipeline successfully added to the git queue')
+
+        except FileNotFoundError as fnfe:
+            self.log.error(f'Unable to locate local DAG file to upload: {pipeline_filepath}: ' + str(fnfe))
+            raise RuntimeError(f'Unable to locate local DAG file to upload: {pipeline_filepath}: {str(fnfe)}') from fnfe
+        except GithubException as e:
+            self.log.error(f'Error uploading DAG to branch {self.branch}: ' + str(e))
+            raise RuntimeError(f'Error uploading DAG to branch {self.branch}: {str(e)}') from e
+
+    def get_github_url(self) -> str:
```

Comment:
Since this is now a somewhat generic utility class could you turn this into a static class method that takes a URL string as an input and returns an URL string? There are two benefits:
- it simplifies the creation of test cases
- it can be invoked by the UI, eliminating the need to implement the same functionality twice (https://github.com/elyra-ai/elyra/issues/1335)

Reply:
updated pr